include/deal.II/grid/grid_tools.h

```diff
@@ -157,10 +158,27 @@ namespace GridTools
    * get the same result using <code>cell-@>measure()</code>, but this
    * function also works for cells that do not exist except that you make it
    * up by naming its vertices from the list.
+   *
+   * Instantiations of this function are available for the <tt>dim !=
+   * spacedim</tt> cases, but these just raise exceptions. They are only
+   * provided as an aid for generic programming.
+   */
+  template <int dim, int spacedim>
```

Comment:
I suggest a very pragmatic approach: Put the template default argument in a preprocessor guard:
```
#ifdef DEAL_II_WITH_CXX11
template <int dim, int spacedim = dim>
#else
template <int dim, int spacedim>
#endif
```

This way we do not forget to put the default argument back in place.
Further, let's face it, 99% of all users (and developers) use C++11 anyway...